### PR TITLE
Migration from Redis to Postgres DB

### PR DESCRIPTION
**Pull request by EnoRage**

### CODE REVIEW

**Comment by hewigovens** [comment]
@vikmeup I remember Danilo had tried PG? no idea why it didn't work for him

File: db/models/subscriptions.go [hunk 1]
```diff
@@ -4,13 +4,14 @@ import "github.com/jinzhu/gorm"
 
 type Subscription struct {
 	gorm.Model
-	GUID string             `gorm:"primary_key:true"`
+	GUID string             `gorm:"type:varchar(256);unique_index"`
```
Comment (vikmeup) [inline db/models/subscriptions.go]: let's use 64 here

File: db/models/subscriptions.go [hunk 2]
```diff
@@ -4,13 +4,14 @@ import "github.com/jinzhu/gorm"
...
 	Data []SubscriptionData `gorm:"many2many:subscription_data;association_foreignkey:subscriptionID;foreignkey:GUID"`
 }
 
 type SubscriptionData struct {
 	gorm.Model
-	SubscriptionID string `gorm:"primary_key:true"`
-	Coin           uint
-	Address        string
+	Subscription     Subscription `gorm:"ForeignKey:UserId; not null"`
+	SubscriptionGUID string       `sql:"index"`
```
Comment (vikmeup) [inline db/models/subscriptions.go]: this would be `SubscriptionID` - as you gonna point to ID and not Guid

File: db/models/subscriptions.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+type Subscription struct {
+	gorm.Model
+	GUID string `gorm:"type:varchar(256);unique_index"`
+	Data []SubscriptionData
+}
+
+type SubscriptionData struct {
+	Subscription     Subscription `gorm:"ForeignKey:UserId; not null"`
+	SubscriptionGUID string       `sql:"index"`
```
Comment (vikmeup) [inline db/models/subscriptions.go]: `SubscriptionGUID` is not needed here I think

File: cmd/observer_notifier/main.go [hunk 4]
```diff
@@ -17,8 +16,8 @@ const (
 )
 
 var (
-	confPath string
-	dbConn   *gorm.DB
+	confPath   string
+	dbInstance *db.Instance
```
Comment (vikmeup) [inline cmd/observer_notifier/main.go]:
```suggestion
	db *db.Instance
```

Comment (EnoRage) [reply cmd/observer_notifier/main.go]: using the same naming as package?

**Pull request merged**